### PR TITLE
feat: enhance keyboard interactions

### LINKED ISSUE #3434: Implement key interactions of `ArrowRight` and `ArrowLeft` for `PinInput`
### Description

Implement key interactions of `ArrowRight` and `ArrowLeft` for `PinInput`

### Problem Statement/Justification

https://github.com/yamada-ui/yamada-ui/issues/1548#issuecomment-2466531238

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

Yes

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by uraaaa24**
Closes #3434

## Description

Implement key interactions of ArrowRight and ArrowLeft for PinInput

### CODE REVIEW

**Comment by cidkumagai** [comment]
@uraaaa24 
I think we need to write a test case for the keybord operation.

**Review by odendayoko — CHANGES_REQUESTED** [review]
@uraaaa24
Thank you for the implementation! I've made two comments, could you please check them?

As @cidkumagai has also mentioned, please add test code.

File: packages/components/pin-input/src/pin-input.tsx [hunk 1]
```diff
@@ -295,23 +312,40 @@ export const PinInput = forwardRef<PinInputProps, "div">(
...
+            focusInputField("next", index)
+          },
+          Backspace: () => {
+            if ((ev.target as HTMLInputElement).value === "") {
+              const prevInput = descendants.prevValue(index, undefined, false)
+
+              if (!prevInput) return
+
+              setValue("", index - 1, false)
+              focusInputField("prev", index)
```
[the remark this answers is not in the document]
Comment (uraaaa24) [reply packages/components/pin-input/src/pin-input.tsx:334]: You are correct, I will fix it that way!

File: packages/components/pin-input/src/pin-input.tsx [hunk 2]
```diff
@@ -211,6 +211,23 @@ export const PinInput = forwardRef<PinInputProps, "div">(
...
+        const input =
+          direction === "next"
+            ? descendants.nextValue(index, undefined, false)
+            : descendants.prevValue(index, undefined, false)
+
+        if (!input) return
+
+        const valueLength = input.node.value.length
+
+        input.node.focus()
```
[the remark this answers is not in the document]
Comment (uraaaa24) [reply packages/components/pin-input/src/pin-input.tsx:225]: It certainly seems like a good idea to include `requestAnimationFrame`.
I will implement it that way!

**Comment by uraaaa24** [comment]
@cidkumagai @uraaaa24 
I have completed the corrections you pointed out.
Thank you for your understanding!

**Review by odendayoko — APPROVED** [review]
@uraaaa24 
Thanks! LGTM🧡

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**